pytorchfi/neuron_error_models.py

```diff
@@ -213,71 +225,67 @@ def _flip_bit_signed(self, orig_value, max_value, bit_pos):
 
         return torch.tensor(new_value, dtype=save_type)
 
     def single_bit_flip_signed_across_batch(self, module, input_val, output):
-        corrupt_conv_set = self.get_corrupt_layer()
-        range_max = self.get_conv_max(self.get_current_layer())
-        logging.info("Current layer: %s", self.get_current_layer())
-        logging.info("Range_max: %s", range_max)
+        corrupt_conv_set = self.corrupt_layer
+        range_max = self.get_conv_max(self.current_layer)
+        logging.info(f"Current layer: {self.current_layer}")
+        logging.info(f"Range_max: {range_max}")
 
         if type(corrupt_conv_set) is list:
             inj_list = list(
                 filter(
-                    lambda x: corrupt_conv_set[x] == self.get_current_layer(),
+                    lambda x: corrupt_conv_set[x] == self.current_layer,
                     range(len(corrupt_conv_set)),
                 )
             )
             for i in inj_list:
                 self.assert_inj_bounds(index=i)
-                prev_value = output[self.corrupt_batch[i]][self.corrupt_dim1[i]][
-                    self.corrupt_dim2[i]
-                ][self.corrupt_dim3[i]]
+                prev_value = output[self.corrupt_batch[i]][self.corrupt_dim[0][i]][
+                    self.corrupt_dim[1][i]
+                ][self.corrupt_dim[2][i]]
 
                 rand_bit = random.randint(0, self.bits - 1)
-                logging.info("Random Bit: %d", rand_bit)
+                logging.info(f"Random Bit: {rand_bit}")
                 new_value = self._flip_bit_signed(prev_value, range_max, rand_bit)
 
-                output[self.corrupt_batch[i]][self.corrupt_dim1[i]][
-                    self.corrupt_dim2[i]
-                ][self.corrupt_dim3[i]] = new_value
+                output[self.corrupt_batch[i]][self.corrupt_dim[0][i]][
+                    self.corrupt_dim[1][i]
+                ][self.corrupt_dim[2][i]] = new_value
 
         else:
-            if self.get_current_layer() == corrupt_conv_set:
-                prev_value = output[self.corrupt_batch][self.corrupt_dim1][
-                    self.corrupt_dim2
-                ][self.corrupt_dim3]
+            if self.current_layer == corrupt_conv_set:
+                prev_value = output[self.corrupt_batch][self.corrupt_dim[0]][
+                    self.corrupt_dim[1]
+                ][self.corrupt_dim[2]]
 
                 rand_bit = random.randint(0, self.bits - 1)
-                logging.info("Random Bit: %d", rand_bit)
+                logging.info(f"Random Bit: {rand_bit}")
                 new_value = self._flip_bit_signed(prev_value, range_max, rand_bit)
 
-                output[self.corrupt_batch][self.corrupt_dim1][self.corrupt_dim2][
-                    self.corrupt_dim3
+                output[self.corrupt_batch][self.corrupt_dim[0]][self.corrupt_dim[1]][
+                    self.corrupt_dim[2]
                 ] = new_value
 
-        self.updateLayer()
```

Comment:
I think it will be better to keep using function calls specifically for `current_layer` manipulation